db/migrate/20230601075835_create_books.rb

```diff
@@ -0,0 +1,11 @@
+class CreateBooks < ActiveRecord::Migration[7.0]
+  def change
+    create_table :books do |t|
+      t.string :name
```

Comment:
we can also specify nullable constraint to the database schema for the attributes that are not allowed to be null

```suggestion
      t.string :name, null: false
```

and also on rails side we can use ActiveRecord Validation for validating the attributes' value before writing it to the database (https://guides.rubyonrails.org/active_record_validations.html)

Reply:
สงสัยนิดหน่อยค่ะว่า ถ้าจะแก้ db สามารถแก้จาก file migrate เก่าได้เลยหรือว่าควรสร้าง migrate file ใหม่อ่อคะ

Reply:
มันจะแบ่งเป็น 2 กรณีครับ
1. ถ้าเรา merge PR ขึ้นไปบน staging/production แล้ว (เท่ากับ database จริง โดน migrate ไปแล้ว ห้าม rollback) เราจะเขียน migration อันใหม่สำหรับแก้ครับ
2. ถ้าเรายังไม่ได้ merge PR (เท่ากับ database ที่โดน migrate คือ database บนเครื่องเรา ซึ่งสามารถ rollback ได้ เพราะไม่มีข้อมูลอะไรสำคัญ) เราสามารถ rollback แล้วแก้ในไฟล์เดิม แล้วค่อย migrate ไปใหม่ได้ครับผม

Reply:
อ่าอาเครค่ะ ขอบคุณค่า🙇‍♀️